app/_kong_plugins/ai-mcp/index.md

```diff
@@ -0,0 +1,220 @@
+
+search_aliases:
+  - ai
+  - llm
+  - artificial
+  - intelligence
+  - language
+  - model
+
+next_steps:
+  - text: Learn about Kong MCP traffic gateway
+    url: /mcp/
+  - text: Learn about {{site.konnect_product_name}} MCP Server
+    url: /mcp/kong-mcp/get-started/
+  - text: Autogenerate MCP tools from a RESTful API
+    url: /mcp/autogenerate-serverless-mcp-tools/
+  - text: Autogenerate MCP tools for Weather API
+    url: /mcp/autogenerate-mcp-tools-for-weather-api/
+---
+The **AI MCP** plugin lets you expose any Kong-managed Service as a [Model Context Protocol (MCP)](https://modelcontextprotocol.io/) server. It acts as a **protocol bridge**, translating between MCP and HTTP so that MCP-compatible clients can call existing APIs without custom server code.
+
+Instead of building an MCP server from scratch, you provide the plugin with an **OpenAPI-compatible schema** for your upstream Service. The plugin uses this schema to understand the available operations and automatically generate MCP tool definitions. This allows you to make *any* HTTP API part of an MCP workflow, while still leveraging Kong’s plugin ecosystem for authentication, traffic control, and observability.
+
+Because the plugin runs directly on Kong AI Gateway, MCP servers are provisioned dynamically on demand. You don’t need to host or scale them separately, and the Gateway’s performance allows a single node to handle hundreds of thousands of requests per second—your upstream APIs will reach their limits before Kong AI Gateway does.
+
+## Why use the AI MCP plugin
+
+The AI MCP bridges the Kong plugin ecosystem with the MCP world, enabling you to bring all of Kong's traffic management, security, and observability capabilities to MCP endpoints:
+
+<!-- vale off -->
+{% table %}
+columns:
+  - title: Use case
+    key: use_case
+  - title: Example
+    key: example
+rows:
+  - use_case: Authentication
+    example: |
+      Apply [OpenID Connect](/plugins/openid-connect/) or the [Key Auth](/plugins/key-auth/) plugin to an MCP Server
+  - use_case: Rate limiting
+    example: |
+      Use [Rate Limiting](/plugins/rate-limiting/) or [Rate Limiting Advanced](/plugins/rate-limiting-advanced) plugins to control MCP request volume
+  - use_case: Observability
+    example: |
+      Add [logging and tracing plugins](/plugins/?category=logging) for full request and response visibility
+  - use_case: Traffic control
+    example: |
+      Apply [request/response transformation plugins](/plugins/?category=transformations) or [ACL policies](/plugins/acl/)
+{% endtable %}
+<!-- vale on -->
+
+## How it works
+
+The AI MCP plugin handles MCP requests by converting them into standard HTTP calls and returning the responses in MCP format. The flow works as follows:
+
+1. Accepts MCP protocol requests from a client.
+2. Parses the MCP tool call and matches it to an OpenAPI operation.
+3. Converts the operation into a standard HTTP request.
+4. Sends the request to the upstream Service.
```

Comment:
In fact, we send the request to Kong again and let Kong proxy it to the upstream service, so that the plugins like rate-limit can run for the generated requests. I am not sure if we need to doc the tech details here. Maybe we will optimize this step in the future.